tests/metrics/genai/test_genai_metrics.py

```diff
@@ -1130,3 +1132,66 @@ def test_make_custom_prompt_genai_metric_validates_input_kwargs():
         custom_judge_prompt_metric.eval_fn(
             input=pd.Series(inputs),
         )
+
+
+def test_log_custom_metric_config_in_make_genai_metric_from_prompt():
+    custom_judge_prompt = "This is a custom judge prompt that uses {input} and {output}"
+
+    custom_metric = make_genai_metric_from_prompt(
+        name="custom",
+        judge_prompt=custom_judge_prompt,
+        greater_is_better=False,
+        parameters={"temperature": 0.0},
+    )
+
+    assert custom_metric.custom_metric_config is not None
+
+    expected_metric_config = {
+        "name": "custom",
+        "judge_prompt": custom_judge_prompt,
+        "model": _get_default_model(),
+        "parameters": {"temperature": 0.0},
+        "aggregations": None,
```

Comment:
`make_genai_metric_from_prompt` has arguments whose values are set within the function. Should we serialize those values? For example, `["mean", "variance", "p90"]` instead of `None` for `aggregations`.

Reply:
the idea here is to log the initial state of these arguments when they are first passed to make_genai_metric_from_prompt call - cuz later during deser we will simply call this function again to construct the EvaluateMetric instance. if the values for "aggregations" or other arguments are changed within the function, it would be covered by re-calling the function itself. 

the `"aggregations": None` here follows the current function default now. If in future mlflow version, we default the argument to a different value, that change will need to modify the test case here.

Reply:
the mlflow_version is logged to the config for this reason. during deser I plan to check if the version matches the ser version in the config. If not I plan to throw a warning to let people know the deser result might not be accurate due to version mismatch.

the underlying assumption I have is that these two functions are probably fairly stable and the scenario for users to deser a metric config from an old artifact is probably rare?

Reply:
Got it, a warning SGTM!